query/src/main/scala/filodb/query/exec/rangefn/AggrOverTimeFunctions.scala

```diff
@@ -584,4 +586,89 @@ class ChangesChunkedFunctionL extends ChangesChunkedFunction with
     }
     changes += longReader.changes(longVect, startRowNum, endRowNum)
   }
-}
\ No newline at end of file
+}
+
+abstract class QuantileOverTimeChunkedFunction(funcParams: Seq[Any],
+                                               var quantileResult: Double = Double.NaN)
+  extends ChunkedRangeFunction[TransientRow] {
+  override final def reset(): Unit = { quantileResult = Double.NaN }
+  final def apply(endTimestamp: Long, sampleToEmit: TransientRow): Unit = {
+    sampleToEmit.setValues(endTimestamp, quantileResult)
+  }
+  def calculateRank(q: Double, counter: Long): (Double, Double, Double) = {
```

Comment:
Shouldn't indexes be Int's, not Doubles?

Reply:
Yes, it should be. I typecast them later. Since rank is a double, when I calculate it returns a Double.